warehouse/slave/worker.go

```diff
@@ -554,6 +558,11 @@ func (w *worker) processMultiStagingFiles(ctx context.Context, job *payloadV2) (
 		jr.cleanup()
 	}()
 
+	// calculate load file name prefix using md5 hash of all staging file locations
+	loadFileNamePrefix := misc.GetMD5Hash(lo.Reduce(job.StagingFiles, func(agg string, item stagingFileInfo, index int) string {
+		return agg + item.Location
+	}, ""))
+
```

Comment:
Can we use a string builder here with bytes pre-allocated? Concatenating such a large string with locations in an append-only fashion seems slow.

https://www.pixelstech.net/article/1715394789-performance-comparison-of-string-concatenation-in-go-language

Reply:
```GO
locationsBuilder := strings.Builder{}
	locationsBuilder.Grow(len(job.StagingFiles) * len(job.StagingFiles[0].Location))
	for _, file := range job.StagingFiles {
		locationsBuilder.WriteString(file.Location)
	}
	loadFileNamePrefix := misc.GetMD5Hash(locationsBuilder.String())
```